### PR TITLE
Fix race condition on initialization of paletteHolder

### PR DESCRIPTION
**Pull request by dimag0g**
There is a race condition between the initialization of two global variables:

    image_palette.cpp:42    PaletteHolder paletteHolder;
    screen.cpp:184          const uint8_t * currentPalette = PALPalette();

If the compiler choses to initialize `currentPalette` first, it uses data from uninitialized `paletteHolder`, typically resulting in all-black palette. This is what happens e.g. with Mingw64 toolchain.

The proposed solution is to access `paletteHolder` via a singleton.

### CODE REVIEW

File: src/engine/image_palette.cpp [hunk 1]
```diff
@@ -36,17 +36,21 @@ namespace
             std::copy_n( kb_pal, paletteSize, gamePalette.begin() );
         }
 
+        static PaletteHolder & singleton()
```
Comment (ihhub) [inline src/engine/image_palette.cpp:39]: Could you please name this method as `instance`? This is done for any newly added code.

File: src/engine/image_palette.cpp [hunk 2]
```diff
@@ -36,17 +36,21 @@ namespace
             std::copy_n( kb_pal, paletteSize, gamePalette.begin() );
         }
```
Comment (ihhub) [inline src/engine/image_palette.cpp:37]: To avoid any future wrongdoings by other developers could we please move this constructor into private section?

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @dimag0g , the change looks good. I left 2 comments here. Can you take a look at them?

**Comment by dimag0g** [comment]
@ihhub All done!

**Review by ihhub — APPROVED** [review]

**Comment by dimag0g** [comment]
@ihhub Thanks for your fixes. Is there anything else for me to do here?

**Comment by ihhub** [comment]
@dimag0g , my apologies. I got busy and had no time to merge it. Thanks a lot for fixing this issue!

**Pull request merged**